frontend/src/hooks/useDebounceSaveNoteAndFetchNoteList.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { Note } from "../common/internal";
+import useAppContext from "./useAppContext";
+
+const DEBOUNCE_SAVE_NOTE_DELAY = 500;
+const DEBOUNCE_FETCH_NOTE_LIST_DELAY = 1500;
+
+const useDebounceSaveNoteAndFetchNoteList = (note: Note) => {
```

Comment:
On a second thought, we should split this into 2 hooks, rather than having it doing multiple things here.